### PR TITLE
move liquid handling out of game.h and game.cpp

### PR DESCRIPTION
**Pull request by KorGgenT**
created a new class liquid_handler that handles liquids, added that as a member to game

#### Summary

```SUMMARY: Infrastructure "move handling liquid functions from game into a member class"```
#### Purpose of change

Better encapsulation of our larger classes. This seemed like a low-hanging fruit, and a chance to categorize other possible classes in game and other large classes.
#### Describe the solution

moves the various handle liquid functions from game into a new class called liquid_handler. there should be no change in functionality. In fact, I did not even try to refactor any of the code at this point.

### CODE REVIEW

**Comment by KorGgenT** [comment]
jenkins rebuild

File: src/game.h [hunk 1]
```diff
@@ -1151,6 +1039,8 @@ class game
         /** How far the tileset should be zoomed out, 16 is default. 32 is zoomed in by x2, 8 is zoomed out by x0.5 */
         int tileset_zoom;
 
+        liquid_handler liquid_handling;
```
Comment (codemime) [inline src/game.h]: There's no point keeping this in `game` as an aggregate: `game` doesn't make any use of it (its environment does).

Comment (KorGgenT) [reply src/game.h]: so even just putting in a namespace then?
all it is is a bunch of functions.

**Comment by codemime** [comment]
Also, there's no point creating a new class either: it has no state. Essentially, it's a bunch of static functions, mostly related to UI.

I'd suggest the following:

- Extract the "core" liquid logic (UI-independent) and distribute it among `map`, `item`, and, possibly, `game`. There may be no such logic, though;
- Whatever remains belongs to UI: put it into the `ui` namespace as a set of static functions (see `game_inventory.h` as an example).

File: src/game.h (hunk 1 above)
Comment (codemime) [reply src/game.h]: Yes, exactly this [I was too slow writing my comment below].

**Comment by KorGgenT** [comment]
i'm going to start with the easy part, and just make its own namespace.
You did say it's mostly UI stuff, but i see that it actually does the liquid handling in there. as far as divvying out that logic, i'll have to give it some thought.

**Pull request merged**